src/goose.rs

```diff
@@ -713,43 +716,61 @@ impl GooseClient {
             weighted_on_start_tasks: Vec::new(),
             weighted_tasks: Vec::new(),
             weighted_on_stop_tasks: Vec::new(),
+            redirected_host: Arc::new(RwLock::new("".to_string())),
             task_request_name: None,
             request_name: None,
             load_test_hash,
         }
     }
 
+    /// A helper to determine which host should be prepended to relative load test
+    /// paths.
+    ///
+    /// The first of these defined will be returned as the prepended host:
+    ///  1. self.redirected_host
+    ///  2. `--host` (host specified on the command line when running load test)
+    ///  3. `GooseTaskSet.host` (default host defined for the current task set)
+    ///  4. `GooseAttack.host` (default host defined for the current load test)
+    async fn get_base_url(&self) -> Url {
+        // If the base url has been redirected, build the URL with it.
+        if let Some(host) = self.get_redirected_host().await {
```

Comment:
You need to call that anyway, so best to store the base_url directly in there.

Reply:
Done